geocodio/data.py

```diff
@@ -171,32 +171,35 @@ def get(self, key):
         Returns an individual Location by query lookup, e.g. address, components dict, or point.
         """
 
-        if isinstance(key, tuple):
-            # TODO handle different ordering
-            try:
-                x, y = float(key[0]), float(key[1])
-            except IndexError:
-                raise ValueError("Two values are required for a coordinate pair")
-
-            except ValueError:
-                raise ValueError("Only float or float-coercable values can be passed")
-
-            key = "{0},{1}".format(x, y)
-        elif isinstance(key, dict):
+        ## Not currently supporting dictionaries in the batch_reverse function
+        # if isinstance(key, tuple):
+        #     # TODO handle different ordering
+        #     try:
+        #         x, y = float(key[0]), float(key[1])
+        #     except IndexError:
+        #         raise ValueError("Two values are required for a coordinate pair")
+
+        #     except ValueError:
+        #         raise ValueError("Only float or float-coercable values can be passed")
+
+        #     key = "{0},{1}".format(x, y)
+        if isinstance(key, dict):
             key = json.dumps(key)
+        elif key in self.keys():
+            return super().get(key)
 
         return self[self.lookups[key]]
```

Comment:
I believe we're still supporting supporting Python 2.7 as of right now. The expression `super()` isn't compatible with Python 2.7 as mentioned [here](https://stackoverflow.com/a/10483204), so for backwards compatibility, it'd need to be `super(LocationCollectionDict, self)` instead. You can test backwards compatibility and any breaking changes using tox.

Reply:
Python 2.7 support is now dropped